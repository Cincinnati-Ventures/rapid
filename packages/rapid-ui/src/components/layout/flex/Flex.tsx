import React, { forwardRef, DetailedHTMLProps, HTMLAttributes } from 'react';
import { RapidStyles } from '../../../utils';

const RAPID_CLASSNAME = 'rapid-flex';

interface FlexProps
	extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
	styles?: string;
}

const Flex = forwardRef<HTMLDivElement, FlexProps>(
	({ styles, children, ...rest }, ref) => {
		const flexStyles = 'flex';

		return (
			<div
				ref={ref}
				{...rest}
				className={RapidStyles(
					styles || rest.className,
					flexStyles,
					RAPID_CLASSNAME,
				)}
			>
				{children}
			</div>
		);
	},
);

export default Flex;
