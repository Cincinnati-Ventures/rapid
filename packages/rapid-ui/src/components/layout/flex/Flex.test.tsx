import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import Flex from './Flex';

describe('Flex', () => {
	it('should render default correctly', () => {
		const { container } = render(<Flex />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with custom styles and children', () => {
		const { container } = render(
			<Flex styles='justify-center items-center space-x-2'>
				<div>Flex 1</div>
				<div>Flex 2</div>
				<div>Flex 3</div>
			</Flex>,
		);
		expect(container).toMatchSnapshot();
	});
});
