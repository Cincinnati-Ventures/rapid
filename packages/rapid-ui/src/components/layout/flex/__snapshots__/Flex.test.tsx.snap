// Vitest Snapshot v1

exports[`Flex > should render correctly with custom styles and children 1`] = `
<div>
  <div
    class="flex justify-center items-center space-x-2 rapid-flex"
  >
    <div>
      Flex 1
    </div>
    <div>
      Flex 2
    </div>
    <div>
      Flex 3
    </div>
  </div>
</div>
`;

exports[`Flex > should render default correctly 1`] = `
<div>
  <div
    class="flex rapid-flex"
  />
</div>
`;
